src/utils/IdealGraphVisualizer/Filter/src/main/java/com/sun/hotspot/igv/filter/FilterChain.java

```diff
@@ -37,60 +38,44 @@
  */
 public class FilterChain implements ChangedEventProvider<FilterChain> {
 
-    private List<Filter> filters;
-    private transient ChangedEvent<FilterChain> changedEvent;
+    private final List<Filter> filters;
+    private final transient ChangedEvent<FilterChain> changedEvent;
+    private final String name;
 
-    private ChangedListener<Filter> changedListener = new ChangedListener<Filter>() {
+    private final ChangedListener<Filter> changedListener = new ChangedListener<Filter>() {
         @Override
         public void changed(Filter source) {
             changedEvent.fire();
         }
     };
 
-    public FilterChain() {
+    public FilterChain(String name) {
+        this.name = name;
         filters = new ArrayList<>();
         changedEvent = new ChangedEvent<>(this);
     }
 
-    public FilterChain(FilterChain f) {
-        this.filters = new ArrayList<>(f.filters);
-        changedEvent = new ChangedEvent<>(this);
+    public FilterChain() {
+        this("");
+    }
+
+    public void sortBy(List<String> order) {
+        filters.sort(Comparator.comparingInt(f -> order.indexOf(f.getName())));
     }
 
     @Override
     public ChangedEvent<FilterChain> getChangedEvent() {
         return changedEvent;
     }
 
-    public Filter getFilterAt(int index) {
-        assert index >= 0 && index < filters.size();
-        return filters.get(index);
-    }
-
-    public void apply(Diagram d) {
-        for (Filter f : filters) {
-            f.apply(d);
-        }
-    }
-
-    public void apply(Diagram d, FilterChain sequence) {
-        List<Filter> applied = new ArrayList<>();
-        for (Filter f : sequence.getFilters()) {
-            if (filters.contains(f)) {
-                f.apply(d);
-                applied.add(f);
-            }
-        }
-
-
-        for (Filter f : filters) {
-            if (!applied.contains(f)) {
-                f.apply(d);
+    public void applyInOrder(Diagram diagram, FilterChain filterOrder) {
```

Comment:
I have also addressed this in [#12955](https://github.com/openjdk/jdk/pull/12955), but I think your solution of sorting the filter list upfront, rather than every time filters are applied, is preferable. I will wait for this PR to be integrated and then exclude the corresponding changes from #12955.

Reply:
sounds good